addons/binding/org.openhab.binding.solarview/src/main/java/org/openhab/binding/solarview/handler/solarviewHandler.java

```diff
@@ -0,0 +1,417 @@
+            logger.info("Exception occurred on socket creation: {}", e.getMessage());
+            energyInformation = null;
+            logger.debug("updateEnergyDataFromServer() return false.");
+            return (false);
+        }
+
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    DataOutputStream outToServer = new DataOutputStream(clientSocket.getOutputStream());
+                    BufferedReader inFromServer = new BufferedReader(
+                            new InputStreamReader(clientSocket.getInputStream()));
+                    logger.trace("updateEnergyDataFromServer() using query string '{}'.", queryString);
+                    outToServer.writeBytes(queryString + SOLARVIEW_REQUEST_EOL);
+                    energyInformation = inFromServer.readLine();
+                    logger.trace("updateEnergyDataFromServer() got response.");
+
+                } catch (IOException e) {
+                    logger.info("Exception occurred during reading on socket: {}", e.getMessage());
+                    energyInformation = null;
+                }
+            }
+        }).start();
+
+        try {
+            Thread.sleep(SERVER_CONNECTION_TIMEOUT_MSECS);
+        } catch (InterruptedException e) {
+            logger.info("Exception occurred during waiting on socket: {}", e.getMessage());
+            energyInformation = null;
+        }
+
+        try {
+            clientSocket.close();
+        } catch (IOException e) {
+            logger.info("Exception occurred during closing socket: {}", e.getMessage());
+            energyInformation = null;
+        }
+        if (energyInformation != null) {
+            energyInformation = energyInformation.replaceAll("^.*\\{", "").replaceAll("\\}.*$", "");
+            logger.trace("energyInformation updated to {}.", energyInformation);
+            logger.trace("updateEnergyDataFromServer() return true.");
+            return (true);
+
+        } else {
+            logger.trace("updateEnergyDataFromServer() return false.");
+            return (false);
+        }
+    }
+
+    /**
+     * Based on the (already stored) responses of the Solarview server
+     * the requested item will be passed back as return value.
+     *
+     * @param channel identifier to query
+     * @return the value of the channel as Decimal or, in case of failure, UNDEF.
+     */
+    private State getChannelValue(String id) {
+        logger.trace("getChannelValue() called for channel {}.", id);
+        int itemIndex;
+        /**
+         * Sample input from Solarview server:
+         * {00,07,06,2017,12,26,0008.2,00169,002467,00024890,00121,000,000.0,000,000.0,000,000.0,000,000.0,00},<
+         *
+         * Explanation:
+         * {WR,Day,Month,Year,Hour,Minute,KDY,KMT,KYR,KT0,PAC,UDC,IDC,UDCB,IDCB,UDCC,IDCC,UL1,IL1,TKK}
+         */
+        switch (id) {
+            case CHANNEL_KDY:
```

Comment:
Hhhhm, should I already rethink about string-to-number mapping? 
From my point of view we would loose the sanity check of wrong/badly passed channel names...
Perhaps I've not got you right.